classes/class-swsales-reports.php

```diff
@@ -366,4 +631,64 @@ public static function ajax_tracking() {
 		echo 'Invalid Report. ';
 		return;
 	}
+
+	/**
+	 * Given 2 SWSales_Sitewide_Sale objects, returns an array with the difference rate between them over several attributes.
+	 *
+	 * @param SWSales_Sitewide_Sale $sitewide_sale_1 A  SWSales_Sitewide_Sale.
+	 * @param SWSales_Sitewide_Sale $sitewide_sale_2 A  SWSales_Sitewide_Sale.
+	 * @return an Array with the markup representing difference rate between them over several attributes.
+	 * @since TBD.
+	 */
+	private static function build_diff_rate_array($sitewide_sale_1, $sitewide_sale_2) {
+		// No need to validate params, we did it before.
+		$diff_rate = [];
+
+		$attributes = [
+			'banner_impressions' => 'get_banner_impressions',
+			'landing_page_visits' => 'get_landing_page_visits',
+			'checkout_conversions' => 'get_checkout_conversions',
+			'revenue' => 'get_sale_revenue',
+		];
+
+		foreach ($attributes as $key => $method) {
+			$value_1 = (float)$sitewide_sale_1->$method();
+			$value_2 = (float)$sitewide_sale_2->$method();
+
+			if ( $value_1 == $value_2 ) {
+				// No change.
+				$diff_rate[$key] = '';
+			} elseif ( empty( $value_2 ) ) {
+				// Avoid divide by 0.
+				$diff_rate[$key] = self::build_rate_markup( 0, true );
+			} elseif ($value_1 > $value_2) {
+				// Growth.
+				$diff = round( ( ( $value_1 - $value_2 ) / $value_2 ) * 100, 2 );
+				$diff_rate[$key] = self::build_rate_markup($diff, true);
+			} else {
+				// Decline.
+				$diff = round( ( ( $value_2 - $value_1 ) / $value_2 ) * 100, 2 );
+				$diff_rate[$key] = self::build_rate_markup($diff, false);
+			}
+		}
+
+		return $diff_rate;
+	}
+
+	/**
+	 * Given a rate and a boolean indicating if it's a growth or decline, returns the markup representing the rate.
+	 *
+	 * @param Obj can be a float or a String $rate The rate to represent.
+	 * @param Boolean $is_growth Indicates if the rate is a growth or decline.
+	 * @return String with the markup representing the rate.
+	 * @since TBD.
+	 */
+	private static function build_rate_markup( $rate, $is_growth ) {
```

Comment:
Needs escaping.

Be sure to note in a comment when these are output later that they were escaped. Maybe even change the var names later to ..._escaped.

Reply:
Did escaping just in the build_rate_markup() method and added comments